docs/content/configuration/auth.md

```diff
@@ -10,6 +10,7 @@ layout: doc_page
 |`druid.escalator.type`|String|Type of the Escalator that should be used for internal Druid communications. This Escalator must use an authentication scheme that is supported by an Authenticator in `druid.auth.authenticationChain`.|"noop"|no|
 |`druid.auth.authorizers`|JSON List of Strings|List of Authorizer type names |["allowAll"]|no|
 |`druid.auth.unsecuredPaths`| List of Strings|List of paths for which security checks will not be performed. All requests to these paths will be allowed.|[]|no|
+|`druid.auth.allowUnauthenticatedHttpOptions`|Boolean|If true, skip authentication checks for HTTP OPTIONS requests. This is needed for certain use cases, such as supporting CORS pre-flight requests. Note that disabling authentication checks for OPTIONS requests will allow unauthenticated users to determine what Druid endpoints are valid (by checking if the OPTIONS request returns a 200 instead of 404), so the authentication checks should not be disabled unless necessary. |false|no|
```

Comment:
Will this leak any info beyond what is available in the docs?

Is there any difference in the response of an OPTIONS request for e.g. `/druid/coordinator/v1/datasources/{dataSource}/intervals` for a real datasource vs a not-real datasource? Or other endpoints like this that include dataSource names in the resource?

Reply:
The "does resource that appears in a URL exist?" leakage won't happen, the responses for the OPTIONS requests are the same:


Valid datasource:
```
curl -i -X OPTIONS http://localhost:8081/druid/coordinator/v1/datasources/wikiticker/intervals
HTTP/1.1 200 OK
Date: Fri, 13 Apr 2018 00:45:34 GMT
Content-Type: application/vnd.sun.wadl+xml
Allow: OPTIONS,HEAD,GET
Last-Modified: Fri, 13 Apr 2018 00:21:06 UTC
Vary: Accept
Content-Length: 711
Server: Jetty(9.3.19.v20170502)

<?xml version="1.0" encoding="UTF-8" standalone="yes"?><application xmlns="http://wadl.dev.java.net/2009/02"><doc xmlns:jersey="http://jersey.java.net/" jersey:generatedBy="Jersey: 1.19.3 10/24/2016 03:43 PM"/><grammars/><resources base="http://localhost:8081/"><resource path="druid/coordinator/v1/datasources/wikiticker/intervals"><method id="getSegmentDataSourceIntervals" name="GET"><request><param xmlns:xs="http://www.w3.org/2001/XMLSchema" name="simple" style="query" type="xs:string"/><param xmlns:xs="http://www.w3.org/2001/XMLSchema" name="full" style="query" type="xs:string"/></request><response><representation mediaType="application/json"/></response></method></resource></resources></application>
````

Invalid datasource:
```
curl -i -X OPTIONS http://localhost:8081/druid/coordinator/v1/datasources/FAKE_DATASOURCES/intervals
HTTP/1.1 200 OK
Date: Fri, 13 Apr 2018 00:45:42 GMT
Content-Type: application/vnd.sun.wadl+xml
Allow: OPTIONS,HEAD,GET
Last-Modified: Fri, 13 Apr 2018 00:21:06 UTC
Vary: Accept
Content-Length: 717
Server: Jetty(9.3.19.v20170502)

<?xml version="1.0" encoding="UTF-8" standalone="yes"?><application xmlns="http://wadl.dev.java.net/2009/02"><doc xmlns:jersey="http://jersey.java.net/" jersey:generatedBy="Jersey: 1.19.3 10/24/2016 03:43 PM"/><grammars/><resources base="http://localhost:8081/"><resource path="druid/coordinator/v1/datasources/FAKE_DATASOURCES/intervals"><method id="getSegmentDataSourceIntervals" name="GET"><request><param xmlns:xs="http://www.w3.org/2001/XMLSchema" name="simple" style="query" type="xs:string"/><param xmlns:xs="http://www.w3.org/2001/XMLSchema" name="full" style="query" type="xs:string"/></request><response><representation mediaType="application/json"/></response></method></resource></resources></application>
```

Reply:
Other examples would be extensions that expose some endpoints when enabled, e.g. `druid-basic-security`:

With extension enabled:
```
curl -i -X OPTIONS http://localhost:8081/druid-ext/basic-security/authorization/ 
HTTP/1.1 200 OK
Date: Fri, 13 Apr 2018 01:15:49 GMT
Content-Type: application/vnd.sun.wadl+xml
Allow: OPTIONS
Last-Modified: Fri, 13 Apr 2018 01:13:22 UTC
Vary: Accept
Content-Length: 4440
Server: Jetty(9.3.19.v20170502)

<?xml version="1.0" encoding="UTF-8" standalone="yes"?><application xmlns="http://wadl.dev.java.net/2009/02"><doc xmlns:jersey="http://jersey.java.net/" jersey:generatedBy="Jersey: 1.19.3 10/24/2016 03:43 PM"/><grammars/><resources base="http://localhost:8081/"><resource path="druid-ext/basic-security/authorization/"><resource path="/db/{authorizerName}/users/{userName}">
...
```

With extension disabled:
```
curl -i -X OPTIONS http://localhost:8081/druid-ext/basic-security/authorization/ 
HTTP/1.1 404 Not Found
Date: Fri, 13 Apr 2018 01:17:30 GMT
Content-Length: 0
Server: Jetty(9.3.19.v20170502)
```

Reply:
In that case I'd suggest wording like this,

> If true, skip authentication checks for HTTP OPTIONS requests. This is needed for certain use cases, such as supporting CORS pre-flight requests. Note that disabling authentication checks for OPTIONS requests will allow unauthenticated users to determine what Druid endpoints are valid (by checking if the OPTIONS request returns a 200 instead of 404), so enabling this option may reveal information about server configuration, including information about what extensions are loaded (if those extensions add endpoints).

Reply:
Sounds good, updated to use that wording